### ISSUE TITLE
Image management in model

### ISSUE THREAD

**Issue by emieza** [lacetans/faceschool#8]
Add image upload to:
* FSUser : profile picture
* FSUser : portrait image
* Post : attached image

Think about two issues:
* where to upload and store images
* how to serve this images in the other views

**Comment by JPRusaki** [comment]
Ok! lets code!

**Comment by emieza** [comment]
One hint to add images to the model:
- https://docs.djangoproject.com/en/1.7/ref/models/fields/#django.db.models.FileField.storage
- https://docs.djangoproject.com/en/1.7/ref/models/fields/#django.db.models.ImageField

For the user profile seems reasonable to use Django backend :)

**Comment by JPRusaki** [comment]
Finished! (I think...)

**Comment by JPRusaki** [comment]
For testing go to /register (register new users), or index (view and create new posts)

**Comment by luschinK** [comment]
I would like to know how an image is stored in django (which directory) to
do the src tag in the posts template​ !

**Comment by JPRusaki** [comment]
you only have to create a new directory under /web named media. You don't have to specify the whole path in the src attribute to the image in the html code, django will do it for us.
for example in latest_posts template:
```javascript
<img src='{{MEDIA_URL}}{{post.image}}' width=40%>
```
the MEDIA URL is configured in the settings file ('/media/') and post.image will be a folder inside /media named /user_images (you don't have to create it). you can try it in your development environment.

**Comment by luschinK** [comment]
ty ! Im already installing Pillow. Nice work !

**Comment by JPRusaki** [comment]
why do you need Pillow? I didn't need that!

**Comment by luschinK** [comment]
Fuck then :/

**Comment by luschinK** [comment]
>
>
>>
web.FSUser.main_image: (fields.E210) Cannot use ImageField because Pillow
is not installed.
HINT: Get Pillow at https://pypi.python.org/pypi/Pillow or run command "pip
install pillow".
web.FSUser.profile_image: (fields.E210) Cannot use ImageField because
Pillow is not installed.
HINT: Get Pillow at https://pypi.python.org/pypi/Pillow or run command "pip
install pillow".
web.Post.image: (fields.E210) Cannot use ImageField because Pillow is not
installed.
HINT: Get Pillow at https://pypi.python.org/pypi/Pillow or run command "pip
install pillow".

**Comment by JPRusaki** [comment]
(0_0) what? are you using django v1.7?

**Comment by luschinK** [comment]
>>> import django
>>> django.VERSION
(1, 7, 1, 'final', 0)

**Comment by luschinK** [comment]
We'll check it later, don' worry.

still ty :)

**Comment by JPRusaki** [comment]
Oh  sorry! my mistake! I thought it was included by default. yes you have to install pillow.

**Comment by luschinK** [comment]
Nice. The next question is :

Did you needed to add columns to sqlite for images?

///
Exception Value: no such column: web_post.image

Exception Location:
/home/funky/env/local/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py
in execute, line 485

**Comment by JPRusaki** [comment]
nope. did you makemigrations and migrate? before that, I made another push yesterday.

**Comment by luschinK** [comment]
Saw it. I updated all today. You forgot an upload_path on Posts. I defined
this one as 'posts_images' and created the directory inside media.
Ill keep trying with makemigatrions.

**Comment by JPRusaki** [comment]
here's the post model in the last push i made:
```javascript
class Post(models.Model):
upload_path = 'users_images'
```
i'ts working for me. you can look that in https://github.com/lacetans/faceschool/blob/image/web/models.py

allusers images (posts, profile and wall) are saved in users_images.

**Comment by JPRusaki** [comment]
you don't have to create any directory inside media django will do it for you.

**Comment by JPRusaki** [comment]
you can try to delete the sqlite db file and try again with makemigrations and migrate.

**Comment by luschinK** [comment]
Perfect. Thanks!

**Comment by JPRusaki** [comment]
Problema detectat!! després d'enviar un formulari de post, si l'usuari refresca la pàgina es torna a eviar el formulari creant posts duplicats!!! He estat buscant i he trobat això: http://en.wikipedia.org/wiki/Post/Redirect/Get

Ho hauríem d'implementar en Django
